Services/UserService/IUserService.cs

```diff
@@ -0,0 +1,46 @@
+﻿using webapi.Dtos.Users;
+using webapi.Models;
+
+namespace webapi.Services.UserService
+{
+    /// <summary>
+    /// Interface for managing user data.
+    /// </summary>
+    public interface IUserService
+    {
+        /// <summary>
+        /// Retrieves all users.
+        /// </summary>
+        /// <returns>A list of all users.</returns>
+        Task<ServiceResponse<List<GetUserDto>>> GetAllUsers();
```

Comment:
why you are mapping serviceResponse here also rename it to serverResponse all kind server response must be done in base controller.

Reply:
Done